src/sentry/api/endpoints/organization_integration_repository_project_path_configs.py

```diff
@@ -1,13 +1,70 @@
 from __future__ import absolute_import
 
+from rest_framework import status, serializers
 
 from sentry.api.bases.organization import OrganizationIntegrationsPermission
 from sentry.api.bases.organization_integrations import OrganizationIntegrationBaseEndpoint
 from sentry.api.serializers import serialize
-from sentry.models import RepositoryProjectPathConfig
+from sentry.api.serializers.rest_framework.base import CamelSnakeModelSerializer
+from sentry.models import RepositoryProjectPathConfig, Project, Repository
 from sentry.utils.compat import map
 
 
+class RepositoryProjectPathConfigSerializer(CamelSnakeModelSerializer):
+    repository_id = serializers.IntegerField(required=True)
+    project_id = serializers.IntegerField(required=True)
+    stack_root = serializers.CharField(required=True)
+    source_root = serializers.CharField(required=True)
+    default_branch = serializers.CharField(required=True)
+
+    class Meta:
+        model = RepositoryProjectPathConfig
+        fields = ["repository_id", "project_id", "stack_root", "source_root", "default_branch"]
+        extra_kwargs = {}
+
+    @property
+    def org_integration(self):
+        return self.context["organization_integration"]
+
+    @property
+    def organization_id(self):
+        return self.org_integration.organization_id
+
+    def validate(self, attrs):
+        # TODO: Needs to be updated for update since the row already exists
+        query = RepositoryProjectPathConfig.objects.filter(
+            project_id=attrs.get("project_id"), stack_root=attrs.get("stack_root")
+        )
+        if query.exists():
+            raise serializers.ValidationError(
+                "Code path config already exists with this project and stack root"
+            )
+        return attrs
+
+    def validate_repository_id(self, repository_id):
+        # validate repo exists on this org and integration
+        repo_query = Repository.objects.filter(
```

Comment:
do we also wanna make sure it's active?

Reply:
@MeredithAnya I mean we could but I don't think that really adds value having that logic. We should be able to do stack trace linking on a disabled repo just fine.

Reply:
But won't the `check_file` request fail? since we won't have permission anymore?

Reply:
Actually I guess if the Repo still has the integration_id we good, thats kind of saying it's active lol

Reply:
@MeredithAnya if the integration is still active, it doesn't matter if the repo is active or not.